tests/glsl-intrinsic/shader-subgroup/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,186 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+//TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl -DTARGET_HLSL
+//TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda -DTARGET_CUDA 
+
+// not testing cpp due to missing impl
+//T-EST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+
+#version 430
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2
+{
+    uint data[];
+} outputBuffer;
+
+#define local_size_x_v 32
+layout(local_size_x = local_size_x_v) in;
+
+__generic<T : __BuiltinLogicalType>
+bool test1Logical() {
+    return true
+#if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
+        && subgroupAnd(T(1)) == T(1)
+        && subgroupOr(T(1)) == T(1)
+        && subgroupXor(T(1)) == T(0) 
+        && subgroupInclusiveAnd(T(1)) == T(1)
+        && subgroupInclusiveOr(T(1)) == T(1)
+        && subgroupInclusiveXor(T(1)) == T(0)
+        && subgroupExclusiveAnd(T(1)) == T(1)
+        && subgroupExclusiveOr(T(1)) == T(1)
+        && subgroupExclusiveXor(T(1)) == T(1)
+#endif // #if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
```

Comment:
I think we will need a comment or more meaningfull macro define about why this block is commented out for HLSL and CUDA.
Something like
```
#if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
// HLSL doesn't have logical operators for subgroup equivalent.
// CUDA side support is not implemented yet.
#define TEST_when_subgroup_logical_operators_are_implemented
#endif
...
#if defined(TEST_when_subgroup_logical_operators_are_implemented)
...
#endif
```